packages/sentry/src/sentry.spec.ts

```diff
@@ -161,3 +161,37 @@ describe('reportErrorToSentry', () => {
     });
   });
 });
+
+describe('when using unsafe fields', () => {
+  it('should redact unsafe fields', () => {
+    let safeFields = removeUnsafeFields({
+      user: {
+        email: 'john@doeland.com',
+      },
+    });
+
+    expect(safeFields.user?.email).toEqual('[Redacted]');
+
+    safeFields = removeUnsafeFields({
+      extra: {
+        action: {
+          payload: {
+            firstName: 'John',
+          },
+        },
+      },
+    });
+
+    // @ts-expect-error
```

Comment:
The `extra` is unknown. I am not sure how I would get around the error `Object is of type 'unknown'` here without jumping through hoops.

Reply:
I think we can just cast it

```ts
(safeFields.extra as MyType)?.action.payload.firstName
```